src/devices_models/devices/common/range_constraint.jl

```diff
@@ -194,13 +366,77 @@ function add_reserve_range_constraints!(
     model::DeviceModel{V, W},
     X::Type{<:PM.AbstractPowerModel},
     feedforward::Union{Nothing, AbstractAffectFeedForward},
+) where {V <: PSY.Component, W <: AbstractDeviceFormulation}
+    # TODO: remove this function? use the impl for dispatch.
+    variable = V()
+    component_type = W
+    array = get_variable(container, variable, component_type)
+    add_reserve_upper_bound_range_constraints_impl!(
+        container,
+        T,
+        array,
+        devices,
+        model,
+        Y,
+        feedforward,
+    )
+    add_reserve_lower_bound_range_constraints_impl!(
+        container,
+        T,
+        array,
+        devices,
+        model,
+        Y,
+        feedforward,
+    )
+end
+
+function add_reserve_range_constraints!(
+    container::OptimizationContainer,
+    T::Type{InputActivePowerVariableLimitsConstraint},
+    U::Type{<:ExpressionType},
+    devices::IS.FlattenIteratorWrapper{V},
+    model::DeviceModel{V, W},
+    X::Type{<:PM.AbstractPowerModel},
+    feedforward::Union{Nothing, AbstractAffectFeedForward},
+) where {V <: PSY.Component, W <: AbstractDeviceFormulation}
+    variable = V()
+    component_type = W
+    array = get_expression(container, variable, component_type)
```

Comment:
also, this will fail 

```suggestion
    array = get_expression(container, U(), component_type)
```